Fix unused but set variable warnings

donny-dont opened the pull request:
#### d6fc4dd71912386c8fdcc171fc5044f28ebbd364
<pre>
Fix unused but set variable warnings
<a href="https://bugs.webkit.org/show_bug.cgi?id=245845">https://bugs.webkit.org/show_bug.cgi?id=245845</a>

Reviewed by Yusuke Suzuki.

Fixed unused but set variable warnings reported by Clang 15.

* Source/JavaScriptCore/runtime/HashMapImplInlines.h:
* Source/JavaScriptCore/runtime/WeakMapImpl.h:
* Source/WTF/wtf/URLParser.cpp:
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
* Source/WebCore/platform/graphics/FontCascade.cpp:
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:

Canonical link: <a href="https://commits.webkit.org/255017@main">https://commits.webkit.org/255017@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec04b802b6886c2288a2814889648adca75b83bc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91016 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35587 "Reverted pull request changes (failure)") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/21547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/100392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/159007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95022 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/34087 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83362 "Built successfully") | [💥 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97146 "An unexpected error occured. Recent messages:Failed to print configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96672 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/77801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/26976 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [💥 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81914 "An unexpected error occured. Recent messages:Failed to print configuration") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/21547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/70012 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/82518 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35166 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/21547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/77520 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32964 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/21547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26703 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3494 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36744 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/77801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/80116 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38672 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/21547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17549 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -8392,7 +8392,6 @@ MediaProducerMediaStateFlags HTMLMediaElement::mediaState() const
     MediaStateFlags state;
 
     bool hasActiveVideo = isVideo() && hasVideo();
-    bool hasAudio = this->hasAudio();
```
I think having these bool here would be easier to read more.
So, let's suppress the warning with `UNUSED_VARIABLE(hasAudio)` in the following `#if`'s `#else` case.

Constellation reviewed (approved):
r=me

donny-dont committed:
Fix unused but set variable warnings
https://bugs.webkit.org/show_bug.cgi?id=245845

Reviewed by Yusuke Suzuki.

Fixed unused but set variable warnings reported by Clang 15.

* Source/JavaScriptCore/runtime/HashMapImplInlines.h:
* Source/JavaScriptCore/runtime/WeakMapImpl.h:
* Source/WTF/wtf/URLParser.cpp:
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
* Source/WebCore/platform/graphics/FontCascade.cpp:
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:

Canonical link: https://commits.webkit.org/255017@main
